### PR TITLE
Add getChipId function and examples

### PR DESCRIPTION
**Pull request by generationmake**
Hi @aentinger 
during bring up of my board with the BMP388 I came across some self test function. One is to check the chip ID of the BMP388. So I added this function to the library. I also added two examples. The first shows only the usage of the getChipId function. The second implements the self-test of Bosch application note https://www.bosch-sensortec.com/media/boschsensortec/downloads/application_notes_1/bst-mps-an006.pdf

### CODE REVIEW

File: examples/BMP388-self-test/BMP388-self-test.ino [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+}
+
+void onExternalEvent()
+{
+  bmp388.onExternalEventHandler();
+}
+
+void onSensorData(double const pressure_hpa, double const temperature_deg)
+{
+//  Serial.print(pressure_hpa);
```
Comment (aentinger) [inline examples/BMP388-self-test/BMP388-self-test.ino]: Please remove the outcommented code from the example.

File: examples/BMP388-self-test/BMP388-self-test.ino [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+/**************************************************************************************
+ * GLOBAL VARIABLES
+ **************************************************************************************/
+
+ArduinoBMP388 bmp388(spi_select,
+                     spi_deselect,
+                     spi_transfer,
+                     onSensorData);
+volatile bool measurement_done;
```
Comment (aentinger) [inline examples/BMP388-self-test/BMP388-self-test.ino]: I suggest immediate initialisation with an initial value, as usually all variables should be assigned a value upon declaration (good defensive coding practive), e.g.
```C++
volatile bool measurement_done = false;
volatile double measurement_pressure_hpa = 0.0;
...
```

File: examples/BMP388-self-test/BMP388-self-test.ino [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+  Serial.print("Chip-ID: 0x");
+  Serial.println(chip_id, HEX);
+  if(chip_id==0x50) Serial.println("Passed!");
+  else Serial.println("Failed!");
+
+  Serial.println("Verify trimming data");
+  /* not possible now */
+
+  Serial.println("measure temperature and pressure");
+  measurement_done=0;
```
Comment (aentinger) [inline examples/BMP388-self-test/BMP388-self-test.ino]: While this statement works due to the internal conversion of `0` to `false` and `!= 0` to `true` it's not a good coding practice. I suggest to directly assign the boolean type `false` here.
```C++
measurement_done = false;
```

File: examples/BMP388-self-test/BMP388-self-test.ino [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+  while(measurement_done==0);
```
Comment (aentinger) [inline examples/BMP388-self-test/BMP388-self-test.ino]: ... and here to check for
```C++
while(!measurement_done) { };
```
In short: Use boolean value constants for boolean variables (everywhere).

File: examples/BMP388-self-test/BMP388-self-test.ino [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+  Serial.println("Passed!");
+}
+
+void loop()
+{
+  uint8_t chip_id;
+  /* Get BMP388 Chip ID*/
+  Serial.println("Verify Chip-ID");
+
+  chip_id=bmp388.getChipId();
```
Comment (aentinger) [inline examples/BMP388-self-test/BMP388-self-test.ino]: This is just a very minor point, but: C++ allows declaration of the variable anywhere in your function before the first usage, they need not be declared at the start of the scope. This is a leftover from C. I'd write instead:
```C++
uint8_t const chip_id = bmp388.getChipId();
```
Same is valid for the other example.

**Review by aentinger — CHANGES_REQUESTED** [review]
Good morning @generationmake :coffee: Thank you very much for your PR, it looks very good already, I've just remarked on a couple of minor issues and kindly ask you to rectify them.

**Comment by aentinger** [comment]
Good morning @generationmake :wave: :coffee: 

When do you have time to finalise this PR? Would love to merge it in rather sooner than later :wink:

**Comment by generationmake** [comment]
Hi @aentinger 
I added the requested changes.
Sorry for the delay.

**Review by aentinger — APPROVED** [review]
Thank you @generationmake :rocket:

**Pull request merged**